prow/github/client.go

```diff
@@ -303,7 +312,14 @@ func (c *Client) requestRaw(r *request) (int, []byte, error) {
 	if !okCode {
 		clientError := ClientError{}
 		if err := json.Unmarshal(b, &clientError); err != nil {
-			return resp.StatusCode, b, err
+			clientError2 := ClientError2{}
+			if err := json.Unmarshal(b, &clientError2); err != nil {
+				return resp.StatusCode, b, err
+			}
+			return resp.StatusCode, b, requestError2{
+				ClientError2: clientError2,
+				ErrorString:  fmt.Sprintf("status code %d not one of %v, body: %s", resp.StatusCode, r.exitCodes, string(b)),
+			}
 		}
 		return resp.StatusCode, b, requestError{
```

Comment:
If you move this logic into a separate `unmarshalClientError` func that returns `error` you could unit test this logic. Currently `TestClientError` doesn't validate this.

This would then look like:
```go
var err error
if !okCode {
  err = unmarshalClientError(b)
}
return resp.StatusCode, b, err
```